### PR TITLE
Micropython buttons support

### PR DESCRIPTION
**Pull request by dwalton76**
For issue #571

### CODE REVIEW

**Comment by dwalton76** [comment]
Huge credit with this patch goes to David's work at https://github.com/dlech/micropython-ev3dev/blob/ev3dev-stetch/src/uev3dev/button.py

**Comment by dwalton76** [comment]
Not sure why the test cases are failing for micropython, they pass when I run them locally via `micropython tests/api_tests.py`.  Will dig some more later...

File: ev3dev2/button.py [hunk 1]
```diff
@@ -337,3 +197,268 @@ def backspace(self):
...
+        raise Exception("micropython button support has not been implemented for '%s'" % platform)
+
+
+    def _test_bit(buf, index):
+        byte = buf[int(index >> 3)]
+        bit = byte & (1 << (index % 8))
+        return bool(bit)
+
+
+    class ButtonBase(ButtonCommon):
```
Comment (WasabiFan) [inline ev3dev2/button.py:223]: What's this for?

**Review by WasabiFan — APPROVED** [review]
Nice! To make sure the API didn't change I [diffed the docs output](https://gist.github.com/WasabiFan/d3c3d7c93ffd0d140655bde815decd6a) and aside from some reordering and wording changes it looks like the API surface is the same. 

So once you figure out what happened to the CI build (maybe an independent package published an update that broke us?) I think this looks OK. Just make sure the general stuff has been tested on both platforms. :+1:

Thanks!

File: ev3dev2/button.py (hunk 1 above)
Comment (dwalton76) [reply ev3dev2/button.py:223]: The InfraredSensor class inherits from this.

Comment (dwalton76) [reply ev3dev2/button.py:223]: yep I tested all the basics on python3 and Micropython. Will look into the CI issues tonight.

**Comment by dwalton76** [comment]
For some reason the `dst` directory isn't getting cleaned up after the first test runs (I have no clue as to why yet) so all of the others fail at `shutil.copytree(src, dst)` because the `dst` directory is already there.  I submitted a PR here:
https://github.com/ev3dev/ev3dev-lang-fake-sys/pull/14

to delete the `dst` directory if it is already there.

**Comment by dwalton76** [comment]
Long story short `shutil.rmtree` is broken in micropython v1.9.4 and that is why the directory isn't cleaned up.  Possible work-around in https://github.com/ev3dev/ev3dev-lang-fake-sys/pull/15

**Comment by dwalton76** [comment]
woohoo finally a clean test run!! :)

**Pull request merged**